src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -790,6 +749,79 @@ public StringBuilder Append(string value, int startIndex, int count)
             }
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            if (value != null && value.Length != 0)
+            {
+                return AppendCore(value, 0, value.Length);
+            }
+            return this;
+        }
+
+        public StringBuilder Append(StringBuilder value, int startIndex, int count)
+        {
+            if (startIndex < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_Index);
+            }
+
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_GenericPositive);
+            }
+
+            if (value == null)
+            {
+                if (startIndex == 0 && count == 0)
```

Comment:
similar, if value is null, shouldn't it throw?
or if not, why does startIndex matter if count is 0?
seems simpler to just throw for null unless there's some scenario for it

Reply:
I kept this for consistency. We are doing similar behavior for other overloads

Reply:
Which ones?

Reply:
https://github.com/Anipik/coreclr/blob/ee5ecc8be2f16746e975640df381f6fe7a4e580d/src/mscorlib/shared/System/Text/StringBuilder.cs#L725
https://github.com/Anipik/coreclr/blob/ee5ecc8be2f16746e975640df381f6fe7a4e580d/src/mscorlib/shared/System/Text/StringBuilder.cs#L619

Reply:
Fair enough.